indico/modules/rb_new/client/js/common/bookings/BookingEditCalendar.jsx

```diff
@@ -44,19 +45,52 @@ class BookingEditCalendar extends React.Component {
         isLoading: false,
     };
 
-    static legendLabels = [
-        {label: Translate.string('Current booking'), style: 'booking'},
-        {label: Translate.string('Invalid occurrence'), style: 'conflicting-candidate'},
-        {label: Translate.string('Cancelled occurrences'), style: 'cancellation'},
-        {label: Translate.string('Pending cancellations'), style: 'pending-cancellation'},
-        {label: Translate.string('Rejected occurrences'), style: 'rejection'},
-        {label: Translate.string('Other bookings'), style: 'other'},
-        {label: Translate.string('New booking'), style: 'available'},
-        {label: Translate.string('Conflicts with new booking'), style: 'conflict'},
-        {label: Translate.string('Blocked'), style: 'blocking'},
-        {label: Translate.string('Blocked (allowed)'), style: 'overridable-blocking'},
-        {label: Translate.string('Not bookable'), style: 'unbookable'}
-    ];
+    transformToLabel = (type) => {
+        switch (type) {
+            case 'candidates':
+                return {label: Translate.string('New booking'), style: 'available', order: 1};
+            case 'bookings':
+                return {label: Translate.string('Current booking'), style: 'booking', order: 2};
+            case 'conflicts':
+                return {label: Translate.string('Conflicts with new booking'), style: 'conflict', order: 3};
+            case 'conflictingCandidates':
+                return {label: Translate.string('Invalid occurrence'), style: 'conflicting-candidate', order: 4};
+            case 'cancellations':
+                return {label: Translate.string('Cancelled occurrences'), style: 'cancellation', order: 5};
+            case 'rejections':
+                return {label: Translate.string('Rejected occurrences'), style: 'rejection', order: 6};
+            case 'pendingCancellations':
+                return {label: Translate.string('Pending cancellations'), style: 'pending-cancellation', order: 7};
+            case 'other':
+                return {label: Translate.string('Other bookings'), style: 'other', order: 8};
+            case 'blockings':
+                return {label: Translate.string('Blocked'), style: 'blocking', order: 9};
+            case 'overridableBlockings':
+                return {label: Translate.string('Blocked (allowed)'), style: 'overridable-blocking', order: 10};
+            case 'nonbookablePeriods':
+            case 'unbookableHours':
+                return {label: Translate.string('Not bookable'), style: 'unbookable', order: 11};
+            default:
+                return undefined;
+        }
+    };
+
+    getLegendLabels = (calendars) => {
+        const legendLabels = [];
+        calendars.forEach((calendar) => {
+            calendar.forEach(({availability}) => {
```

Comment:
This seems very similar to [some other bit](https://github.com/indico/indico/pull/3900/files#diff-ba6397d7aaad7dd18fb13f6c1a947fe3R314) as well as many others after. Maybe it can be moved out to its own function?